### PR TITLE
[Feature]Support SELECT SQL_NO_CACHE

### PR DESCRIPTION
**Pull request by arthur-zhang**
Support SQL_NO_CACHE grammer to make some sql tools work

## Proposed changes

Some sql tool set SQL_NO_CACHE when export data, eg the famouse Querious tool on mac

image

Support SQL_NO_CACHE grammer

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

### CODE REVIEW

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -3509,6 +3509,11 @@ select_hints_list ::=
     :}
     ;
 
+sql_no_cache ::=
```
Comment (yangzhg) [inline fe/fe-core/src/main/cup/sql_parser.cup:3512]:
```suggestion
opt_sql_no_cache ::=
```

Comment (arthur-zhang) [reply fe/fe-core/src/main/cup/sql_parser.cup:3512]: ok

Comment (xinghuayu007) [reply fe/fe-core/src/main/cup/sql_parser.cup]: Doris supports sql_cache and partition_cache. Sql_no_cache grammar means selecting from table with disabled cache. There are no any operation to this grammar. Why not disable sql_cache/partition_cache?

**Comment by yangzhg** [comment]
Please resolve the conflicts

**Pull request closed without merging**